runtime/src/master.rs

```diff
@@ -231,13 +237,13 @@ impl<T: Trait> Module<T> {
             ensure!(valid, MasterError::<T>::BadSig);
         }
 
-        // check/execute
-        proposal
+        // execute call and collect dispatch info to return
+        let dispatch_info = proposal
             .clone()
             .dispatch(system::RawOrigin::Root.into())
             .map_err(|e| e.error)?;
```

Comment:
If an authorized proposal fails, that is not the fault of the submitter. If we short circuit here on error, the submitter will still pay the full price of the extrinsic when the proposal fails.

Also, this is not related to the PR, but it's probably best if round number gets incremented even on failed execution. This is considering the fact that invalid transactions are still included in the block.

An alternative to short-circuiting is to emit an event like in the [sudo pallet](https://github.com/paritytech/substrate/blob/8a35a14c2dd91bc5de77366f45a06fe45bdfd1d6/frame/sudo/src/lib.rs#L140).